### PR TITLE
Move chunking strategies from SCEPA pipeline to database-builder-libs

### LINKED ISSUE #24: Move the chunking strategy of 1 summary + n-number of points to database-builder-libs
## Description

As a developer, I want to implement and move a chunking strategy that creates "1 summary + n-number of points" to database-builder-libs, so that documents can be processed into a structured format with a summary followed by key points for better semantic search and retrieval.

## Requirements

1. [x] Design the summary + points chunking strategy interface

2. [x] Implement the SummaryPointsChunkingStrategy class
   - Create a class that extracts one summary chunk from document sections
   - Implement logic to extract n-number of sections as separate chunks
   - Add configuration for summary extraction method (first section, abstract section, or LLM-generated)
   - Add configuration for points extraction (bullet points, numbered lists, or LLM-extracted)

3. [x] Update TextChunkExtractor to use strategies
   - Refactor existing TextChunkExtractor to use strategy pattern
   - Maintain backward compatibility with current section-based approach
   - Allow strategy injection via constructor

4. [x] Add  unit tests
   - Test summary extraction from various document formats
   - Test points extraction with different configurations
   - Test edge cases (no summary, fewer points than requested)
   - Test chunk metadata accuracy

5. [x] Document the chunking strategy
    - Explain the benefits of summary + points approach
    - Document configuration options
    - Provide best practices for different document types
    - Include examples of when to use this strategy vs others

### PR DESCRIPTION
**Pull request by Hoog-V**
# Chunking strategies

To provide multiple ways of preparing the text for the rag embedding, e.g. sliding window, summary+sections(currently used), or n-fixed sized points. I made some strategy classes which when dropped into the document pipeline, automatically chunk the documents accordingly.

When merged this issue should close #24 

## Autogenerated

This pull request significantly restructures and expands the documentation for the document parsing and chunking pipeline, shifting the focus from embedding utilities to a more modular and detailed explanation of document parsing, supported formats, error handling, and chunking strategies. The changes introduce new documentation files, update the navigation structure, and provide practical examples for users, making the docs clearer and more actionable.

**Documentation Restructuring and Expansion:**

* The embedding-focused documentation (`03_embedding_utilities.md`) has been removed and replaced with new, more detailed documentation on text extraction (`03_text_extract_utilities.md`) and chunking strategies (`08_chunking_strategy.md`). These new docs cover the design, usage, configuration, supported formats, error handling, and return types for `DocumentParserDocling` and all chunking strategies. [[1]](diffhunk://#diff-469204fbe48e4210f0618efd651e8227b724eb78d22a9848dcb6d2c6b6a0ad30L1-L36) [[2]](diffhunk://#diff-53cb334c3e309339af791ef2832f607623ea4e313a05b90390aa18583255e663R1-R124) [[3]](diffhunk://#diff-367ed1392941221199f3bf9634810aacdae8e14f4f47beb954a72ae274b425e3R1-R194)

* The navigation in `mkdocs.yml` is updated to reflect the new documentation structure, ensuring users can easily find content on text extraction and chunking strategies.

**Practical Examples and Thematic Grouping:**

* The main example guide (`02_examples.md`) is overhauled to provide hands-on usage for document parsing (with `DocumentParserDocling`), error handling, supported formats, and all chunking strategies, replacing the previous vectorization example. This includes clear code samples and explanations for each step. [[1]](diffhunk://#diff-de71d2c296c703766efa53975ea795665ff20313af8142811a5794752a709605L8-R9) [[2]](diffhunk://#diff-de71d2c296c703766efa53975ea795665ff20313af8142811a5794752a709605L92-R347)

**Dependency Update:**

* Adds `pypdf>=4.0.0` to the documentation dependencies, likely to support PDF processing in the new examples and Docling backend.

### CODE REVIEW

File: src/database_builder_libs/models/chunk_strategy.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+# Type alias for the raw section tuples produced by TextStructureExtractor.
+# Each entry is (section_title, section_text, list_of_dataframes).
+RawSection = tuple[str, str, list[Any]]
+
+
+# --------------------------------------------------------------------------- #
+# Interface                                                                    #
+# --------------------------------------------------------------------------- #
+
+class ChunkingStrategy(ABC):
```
Comment (SimonDingemans) [inline src/database_builder_libs/models/chunk_strategy.py]: Would like this to be `AbstractChunkingStrategy` to be consistent with other abstract class naming

File: tests/database_builder_libs/util/chunk/test_chunk_strategies.py [hunk 2]
```diff
@@ -0,0 +1,321 @@
+from typing import Any, List
```
Comment (SimonDingemans) [inline tests/database_builder_libs/util/chunk/test_chunk_strategies.py]: I would very much appreciate having this split up into multiple files according to the src files

File: tests/database_builder_libs/util/extract/test_document_parser_docling.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    pdf = Path(__file__).parent / "data" / "ProjectProposalTest.pdf"
+    return pdf.read_bytes()
+
+
+# --------------------------------------------------------------------------- #
+# Unit tests (mocked converter)                                                #
+# --------------------------------------------------------------------------- #
+
+def test_parse_stream_returns_parsed_document(parser):
+    fake_doc = MagicMock(spec=DoclingDocument)
```
Comment (SimonDingemans) [inline tests/database_builder_libs/util/extract/test_document_parser_docling.py]: We continuously do this, can we make the fake_doc a fixture?

**Review by SimonDingemans — CHANGES_REQUESTED** [review]
Looks good to me for the content. But I would like some changes in how it is implemented.

File: src/database_builder_libs/models/chunk_strategy.py (hunk 1 above)
Comment (Hoog-V) [reply src/database_builder_libs/models/chunk_strategy.py]: Ok. renamed it

File: tests/database_builder_libs/util/chunk/test_chunk_strategies.py (hunk 2 above)
Comment (Hoog-V) [reply tests/database_builder_libs/util/chunk/test_chunk_strategies.py]: Fixed, made it separate files. It is now indeed more maintainable

File: tests/database_builder_libs/util/extract/test_document_parser_docling.py (hunk 3 above)
Comment (Hoog-V) [reply tests/database_builder_libs/util/extract/test_document_parser_docling.py]: I made it a fixture

**Comment by Hoog-V** [comment]
Okay have now tested it in the SCEPA pipeline app, and this PR is drop-in replacement by now. So it should now be ready to review.

**Review by SimonDingemans — APPROVED** [review]
